kevin_li/week01/day5/MTA/js/main.js

```diff
@@ -0,0 +1,223 @@
+// // "You must travel through the following stops on the N line: 34th, 28th, 23rd, Union Square."
+// // "Change at Union Square."
+// // "Your journey continues through the following stops: 23rd, 28th, 33rd."
+// // "7 stops in total."
+// There are 3 subway lines:
+// The N line has the following stops: Times Square, 34th, 28th, 23rd, Union Square, and 8th
+// The L line has the following stops: 8th, 6th, Union Square, 3rd, and 1st
+// The 6 line has the following stops: Grand Central, 33rd, 28th, 23rd, Union Square, and Astor Place.
+// All 3 subway lines intersect at Union Square, but there are no other intersection points. (For example, this means the 28th stop on the N line is different than the 28th street stop on the 6 line, so you'll have to differentiate this when you name your stops in the arrays.)
+// Tell the user the number of stops AND the stops IN ORDER that they will pass through or change at.
+// Hints:
+//
+// Work out how you would do it on paper first! Then start to explain that process in Javascript.
+// Get the program to work for a single line before trying to tackle multiple lines.
+// Don't worry about prompting the user for input. Hard code some values to get it working. You can use prompt() later to make it more interactive.
+// Consider diagramming the lines by sketching out the subway lines and their stops and intersection.
+// The key to the lab is finding the index positions of each stop. (hint: indexOf())
+// Make sure the stops that are the same for different lines have different names (i.e. 23rd on the N and on the 6 need to be differentiated)
+
+
+// Arrays
+var lineN = ['Times Square', '34th', '28th', '23rd', 'Union Square', '8th']
+var lineL = ['8th', '6th', 'Union Square', '3rd', '1st']
+var line6 = ['Grand Central', '33rd', '28th', '23rd', 'Union Square', 'Astor Place']
+
+// // Index
+// // line.indexOf(stop)
+//
+// // Number of total stops to travel within a line
+// var stopTotal = function (startStop, endStop, line) {
+//   var startStopIndex = line.indexOf(startStop);
+//   var endStopIndex = line.indexOf(endStop);
+//   return Math.abs(startStopIndex - endStopIndex);
+// };
+//
+// console.log(stopTotal('8th', '1st', lineL));
+//
+// // Stops to travel within a line
+// var stopTravel = function (startStop, endStop, line) {
+//   var i = line.indexOf(startStop);
+//   var j = line.indexOf(endStop);
+//   if (i < j) {
+//     return line.slice(i+1,j+1).join(', '); // need to use an if statement to make it works when travelling backwards.
+//   }
+//     else if (i > j) {
+//       line = line.reverse(line);
+//       return line.slice(j+1,i+1).join(', ');
+//   }
+// };
+//
+// console.log(stopTravel('6th', '1st', lineL));
+// console.log(stopTravel('1st', '8th', lineL));
+
+// Line name
+var lineName = function (line) {
+  if (line === lineN) {
+    return 'N line';
+  } else if (line === lineL) {
+    return 'L line';
+  } else if (line === line6) {
+    return '6 line';
+  }
```

Comment:
If you used an object as your main data structure, with the line names as keys, and the array of stations on the line as the key's value, you wouldn't have to do all this translation of variable names into strings.... much easier.